File: app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/patterns/stops/either/EitherStopState.java
package rectangledbmi.com.pittsburghrealtimetracker.patterns.stops.either;

import io.reactivex.Flowable;
import io.reactivex.Observable;
import rectangledbmi.com.pittsburghrealtimetracker.patterns.PatternViewModel;

/**
 * <p>Workaround for emitting a zoom or selection state from {@link Observable#merge(Iterable)}
 *    in {@link PatternViewModel#getStopRenderRequests(Flowable)} </p>
 * <p>Created by epicstar on 10/9/16.</p>
 *
 * @param <T> the class type that is encapsulated in this interface
 *
 * @author Jeremy Jao
 * @author Michael Antonacci
 * @since 78
 */
public interface EitherStopState<T> {
    T value();
}

File: app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/patterns/stops/either/EitherStopState.kt
package rectangledbmi.com.pittsburghrealtimetracker.patterns.stops.either

import io.reactivex.Observable
import rectangledbmi.com.pittsburghrealtimetracker.patterns.PatternViewModel
import rectangledbmi.com.pittsburghrealtimetracker.patterns.stops.StopRenderState

/**
 *
 * Sum type for emitting a zoom or selection state from [Observable.merge]
 * in [PatternViewModel.getStopRenderRequests]
 *
 * Created by epicstar on 10/9/16.
 *
 *
 *
 * @author Jeremy Jao
 * @author Michael Antonacci
 * @since 78
 */

sealed class EitherStopState
data class FullStopSelectionState(val stopRenderStateMap: Map<Int, StopRenderState>) : EitherStopState()
data class MapState(val shouldStopsBeVisible: Boolean) : EitherStopState()
